gridded_etl_tools/utils/metadata.py

```diff
@@ -683,6 +684,13 @@ def encode_vars(self, dataset: xr.Dataset) -> xr.Dataset:
                         "units": None,
                     }
                 )
+
+        # Encrypt variable data if requested
+        if self.encryption_key is not None:
+            dataset[self.data_var()].encoding["filters"] = [
+                EncryptionFilter(self.encryption_key)
+            ]
```

Comment:
Nitpicking a bit, but I think we should check for existing filters and append if there are any instead of always overwriting. Maybe not strictly necessary for code in current state, but helps future proof.

Reply:
0d581f0

True. Currently, we know that `remove_unwanted_fields` is going to initialize that for us, but best not to assume.